dlpy/layers.py

```diff
@@ -233,6 +292,25 @@ def rnn_summary(self):
         return pd.DataFrame([[self.layer_id, self.name, self.type, self.activation, self.output_size]],
                             columns=['Layer Id', 'Layer', 'Type', 'Activation', 'Output Size'])
 
+    def _assert_inputs(self, inputs):
+        '''
+        Check if inputs are tensor, inputs tensor are compatible in term of shape and layer property
+
+        inputs: a list of Tensor object
+
+        '''
+        # TODO: check if input tensors are compatible with each type of the layer.
+        for input in inputs:
+            if not isinstance(input, Tensor):
+                raise ValueError('Layer {} is called with an input that isn\'t a tensor object'.format(self.name))
+
+
+def Input(n_channels=None, width=None, height=None, name=None, nominals=None, std=None, scale=None,
```

Comment:
we already have inputlayer, why we need another class?

Reply:
Input returns Tensor type object and InputLayer is a Layer class. When a layer is called, it requires tensor(s) rather than Layer object(s). For example, **Tensor** = Input(...); **Tensor** = Dense(n = 10)(**Tensor**)